### PR TITLE
fix(paginator) allow composite pagination keys with float columns

### PR DESCRIPTION
**Pull request by devx-opensource**
This fix allows advanced pagination keys where the unique column
key doesn't have to be called `id` and can be specified through
a new attribute called `paginationUniqueKey`

`paginationUniqueKey` defaults to `id`

----

Hi,

I had a case where the pagination was wrong with the same entry showing up in the next page. 
The issue was two fold:  

- `decode` was transforming the value stored in the cursor with `parseInt` instead of `parseFloat`. This alone would cause bugs with a float column as the query would look for items greater or lower than the integer value, hence missing entries. 

> Example:

> paginationKeys: [balance, id]
> order: DESC
> first page has the following last entry (balance: 1.55, id: 2)
> query for next page would have balance < 1 instead of balance < 1.55
> ==> missing all potential entries between 1 and 1.55

- `buildCursorQuery` had an incorrect behavior regarding composite pagination keys. 

In a cursor-based pagination, the order of the keys used matters to have a correct ordering.

First, we should always have a key matching a column where values are unique and orderable (`id` is commonly used when it's an auto-increment integer column). This is the responsability to the developper to set a proper one. 

Second, the paginationKeys doesn't always start with that "unique key column". If I want to paginate a set of Users by their account balance, the paginationKeys will be set to ['balance', 'id']. In that case the buildCursorQuery would fail in making a correct SQL query. By correct I mean a working pagination.

This pull request introduces a new optional attribute `paginationUniqueKey` that allows a developer to define the key to use as the "unique key column" which might be necessary in case of composite pagination keys. By default it's set to 'id'.

To sum up:
- [X] Allow float column to be used 
- [X] Fix query in case of composite pagination keys not starting with `id`
- [X] New test case 
- [X] No breaking changes

### CODE REVIEW

**Review by benjamin658 — APPROVED** [review]
LGTM

**Comment by benjamin658** [comment]
Thanks for your contribution, the new feature looks good.

I will release the change in the next minor bump.
